scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/AddDataFileController.java

```
@@ -0,0 +1,86 @@
+package edu.asu.diging.scriptoocloud.web;
+
+import edu.asu.diging.scriptoocloud.core.exceptions.DataFileStorageException;
+import edu.asu.diging.scriptoocloud.core.model.impl.DataFile;
+import edu.asu.diging.scriptoocloud.core.service.IDataFileService;
+import edu.asu.diging.scriptoocloud.core.service.IDatasetService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import java.io.IOException;
+import java.security.Principal;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+@Controller
+@PropertySource("classpath:/config.properties")
+public class AddDataFileController {
+
+    @Value("${pageSize}")
+    private String paginationSize;
```

I believe you can make this an integer and then you won't have to parse it later. Also, this way it should throw an exception on startup if `pageSize` has an invalid value and not when a request is made.